cleanlab_studio/internal/dataset_source/pandas_dataset_source.py

```diff
@@ -0,0 +1,26 @@
+import io
+import os
+from typing import IO, Any
+import pandas as pd
+
+from .dataset_source import DatasetSource
+
+
+class PandasDatasetSource(DatasetSource):
+    def __init__(self, *args: Any, df: pd.DataFrame, name: str, **kwargs: Any):
+        super().__init__(*args, **kwargs)
+        self.name = name
+        self._fileobj = self._init_fileobj_from_df(df)
+        self.file_size = self._get_size_in_bytes()
+        self.file_type = "text/csv"
+
+    def _init_fileobj_from_df(self, df: pd.DataFrame) -> IO[bytes]:
+        fileobj = io.BytesIO()
+        df.to_csv(fileobj)
+        fileobj.seek(0)
+        return fileobj
```

Comment:
It's worth noting that w/ Pyspark, we'll have to write to a file (by supplying a path) -- not to a stream. It may make sense to unify things a bit by having this write to a tmpfile (and doing the same for pyspark when we build that out)

Reply:
Note to self: test on windows after rewriting this to use tempfile (we had issues with tempfiles on windows not behaving the same way previously)